userspace/libsinsp/sinsp_resource_utilization.cpp

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright (C) 2023 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+#include <cmath>
+#include "sinsp_resource_utilization.h"
+#include <sys/times.h>
+#include <sys/stat.h>
+#include "utils.h"
+#include <sinsp.h>
+#include "sinsp_public.h"
+#include <sys/sysinfo.h>
+
+
+struct sinsp_resource_utilization;
+
+void libsinsp::resource_utilization::get_rss_vsz_pss_memory(uint32_t &rss, uint32_t &vsz, uint32_t &pss)
+{
+	char filename[1024];
```

Comment:
[nitpick] Do we need 1K characters for this? Looks like at most it will only hold 100 characters since it is paths under `/proc/pid/.

On the other hand, we might want to use something like `FILENAME_MAX`, which is going to be way bigger than 1K but it's define that give addition meaning to what the buffer will be holding.